File: example/java/jimmer-cloud/model/src/main/java/org/babyfish/jimmer/example/cloud/model/Author.java
package org.babyfish.jimmer.example.cloud.model;

import org.babyfish.jimmer.example.cloud.model.common.BaseEntity;
import org.babyfish.jimmer.sql.*;

import java.util.List;

@Entity(microServiceName = "author-service")
public interface Author extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    @Key
    String firstName();

    @Key
    String lastName();

    Gender gender();

    @ManyToMany(mappedBy = "authors")
    List<Book> books();
}

File: example/kotlin/jimmer-cloud-kt/model/src/main/kotlin/org/babyfish/jimmer/example/cloud/kt/model/Author.kt
package org.babyfish.jimmer.example.cloud.kt.model

import org.babyfish.jimmer.example.cloud.kt.model.common.BaseEntity
import org.babyfish.jimmer.sql.*

@Entity(microServiceName = "author-service")
interface Author : BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    @Key
    val firstName: String

    @Key
    val lastName: String

    val gender: Gender

    @ManyToMany(mappedBy = "authors")
    val books: List<Book>
}